### ISSUE TITLE
Distillery has left over fluid that doesn't process. Causing massive headache.

### ISSUE THREAD

**Issue by GrimmDemon** [BuildCraft/BuildCraft#4482]
The distillery for refining crude oil and its' byproducts is not processing leftover units. I am using the thermal series pipes to send fluid that can be distilled, crude oil, distilled oil, etc. to the distilleries from tanks. The distilleries are setup to process multiple fluids during a single 'set', to be more exact, converting crude oil to distilled oil to mixed light fuel and then gaseous fuel. The whole process is setup to use searing fluid then hot fluid then cool fluid, as to reduce complexity of the total refinery complex, however the amount of distilleries are abnormal at most and do not allow for fluids to be evenly spread across all of them, the end result is left over fluid in the inlet tank of the distillery, which cannot be pumped out or distilled, which in turn will not allow a new fluid to be pumped in. This is causing a massive headache for me as every time I encounter this issue, I need to destroy the distilleries that are affected, leaving me with useless "fragile shards" of the remnants that didn't want to process, usually 0,003 mB of oil, etc.
I honestly don't know if this is a bug, or if it's designed like this, but It's really frustrating. It takes a long time to recreate, the way I'm doing it, but if you want to recreate it, it's probably best to place one bucket of "Heavy Oil" in a distillery and power it up, there will be left over oil that won't process.

**Comment by GrimmDemon** [comment]
Forgot to mention that I am using the latest Buildcraft release that is available to download from curseforge.

**Comment by AlexIIL** [comment]
Originally the distiller worked like this, where it only accepted fluids in exact multiples of what could be processed, and refused to allow the rest to be inserted. However this caused issues itself, for example #3709 and #3780, so I changed it to accept any amount.

Essentially I don't really want to change it back, so I can only really recommend separating out the system into individual distilleries, which only process a single fluid,

**Comment by GrimmDemon** [comment]
Makes sense. I should of probably expected issues to arise from using a single row of distilleries to refine multiple fluids. A tank of 20 million buckets doesn't quite divide between 26 distilleries.

**Comment by GrimmDemon** [comment]
I just solved my own conundrum by accident. I realized that the distilleries refine fluid at 0.01 mB/t, but the thermal series pipes transfer fluid at 0.001 mB/t, they're efficient, but not fully compatible with the distilleries, so all I had to do was use BC pipes, which 'synchronize' with the distilleries and pump only 0.01 mB/t, effectively nullifying the problem that I had with leftover fluids in the input tank. It would, however, be a good idea to allow the fluids from the input tank to be pumped back out with the correct plumbing, of course, just in case. I don't particularly like using the BC pipes because of the backflow problem, but the do come in handy, sometimes.

**Comment by GrimmDemon** [comment]
I just need to correct my here.
the conversion rate is actually 0.040mB for all distillable fluids with heavy oil being an exception at 0.030 mB. And it isn't per tick as I've noticed, the processing speed is dependent on the temperature of the fluid.
I had to use cheat engine to manipulate the fluids in their containers to achieve tenths of a bucket, or 10mB.
